qa/rpc-tests/test_framework/util.py

```diff
@@ -352,6 +356,28 @@ def start_node(i, dirname, extra_args=None, rpchost=None, timewait=None, binary=
 
     return proxy
 
+def assert_start_raises_init_error(i, dirname, extra_args=None, expected_msg=None):
+    log_stderr = tempfile.SpooledTemporaryFile(max_size=2**16)
+    try:
+        node = start_node(i, dirname, extra_args, stderr=log_stderr)
+        stop_node(node, i)
+        if expected_msg is None:
+            assert_msg = "bitcoind should have exited with an error"
+        else:
+            assert_msg = "bitcoind should have exited with expected error " + expected_msg
+        raise AssertionError(assert_msg)
+    except AssertionError as e:
+        raise
+    except Exception as e:
+        assert('bitcoind exited' in str(e)) #node must have shutdown
```

Comment:
minor nit: assert is a statement, not a function. This should be:

assert 'bitcoind exited' in str(e)